exonum-java-binding-core/rust/src/storage/proof_map_index.rs

```diff
@@ -163,15 +173,75 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
     let res = panic::catch_unwind(|| {
         let key = convert_to_key(&env, key)?;
         env.ensure_local_capacity(512)?;
-        let _proof = match *utils::cast_handle::<IndexType>(map_handle) {
+        let proof = match *utils::cast_handle::<IndexType>(map_handle) {
             IndexType::SnapshotIndex(ref map) => map.get_proof(key),
             IndexType::ForkIndex(ref map) => map.get_proof(key),
         };
-        unimplemented!("ECR-1350")
+        let unchecked_entries = proof.proof_unchecked();
+        let branches: Vec<_> = unchecked_entries.into_iter()
+            .map(|(path, hash)| create_java_map_proof_entry_branch(&env, path, hash))
+            .collect::<Result<Vec<_>>>()?;
+        let checked_proof = proof.check().unwrap();
```

Comment:
I'd probably add a comment explaining that checking a proof is a temporary measure.